src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildWrapper.java

```diff
@@ -549,16 +564,12 @@ private static String getCurrentHostName() {
     /**
      * @return the port to be used
      */
-    private int getPort() {
+    private int getPort(Map<String, String> envVars) {
         if (StringUtils.isNotBlank(seleniumPort) && !seleniumPort.equals("0")) {
             Matcher matcher = ENVIRONMENT_VARIABLE_PATTERN.matcher(seleniumPort);
             if (matcher.matches()) {
                 String variableName = matcher.group(1);
-                String value = System.getenv(variableName);
-                if (value == null) {
-                    value = "0";
-                }
-                return Integer.parseInt(value);
+                return MapUtils.getInteger(envVars, variableName, 0);
```

Comment:
@oleg-nenashev thanks!

I thought: ```EnvVars env = new EnvVars();
try {
     env = build.getEnvironment(listener);
}```

... would get the appropriately cascaded system variables. Can you point me @ an example of properly using the EnvVars API as you suggest?